### PR TITLE
Implement custom redis transaction-like support

### PR DESCRIPTION
**Pull request by Leodau**
## What does this PR do ?

Implements a memoryStorage.**mexecute** action, that receives multiple commands and executes them _directly_ in one go, using the underlying redis transactional support.

```
{
    "controller": "ms",
    "action": "mexecute",
    "body": {
    "actions": [
        { "action": "del", "args": { "_id": "#list" } },
        { "action": "sadd", "args": { "_id": "#list", "body": { "members": ["a", "b"] } } }
    ]
}
```

- [x] Functional & Unit tests
- [x] Implent mexecute in redis base client wrapper
- [x] Expose mexecute in memoryStorage controller route
- [x] Docs
- [x] Match Kuzzle Lvl API

### Boyscout

- One-variable-per-declaration on the **memoryStorage** controller

### CODE REVIEW

File: lib/service/cache/redis.js [hunk 1]
```diff
@@ -208,6 +208,22 @@ class Redis extends Service {
...
+  mexecute (commands) {
+    if (!Array.isArray(commands) || commands.length === 0) {
+      return Bluebird.resolve([]);
+    }
+
+    return new Bluebird(async (resolve) => {
+      const pipeline = this._client.multi(commands);
+      resolve(pipeline.exec());
+    });
+  }
```
Comment (scottinet) [inline lib/service/cache/redis.js:225]: `pipeline.exec` returns a Promise from what I understand, so this should work while being much simpler:

```suggestion
  mexecute (commands) {
    if (!Array.isArray(commands) || commands.length === 0) {
      return Bluebird.resolve([]);
    }

    return this._client
      .multi(commands)
      .exec();
  }
```

**Comment by Aschen** [comment]
Discussion: Should we put this method in the bulk controller?

**Comment by scottinet** [comment]
I disagree. The `bulk` controller, as its name suggests, is made for en-masse imports, bypassing most of Kuzzle's features.
Here, this route aims at providing an access to Redis' [multi](https://redis.io/commands/multi), which allows executing multiple actions as a single, atomic one. So I think that its place is really in the `ms` controller.

File: doc/2/api/controllers/memory-storage/mexecute/index.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+---
+code: true
+type: page
+title: mexecute
+---
+
+# mexecute
```
[the remark this answers is not in the document]
Comment (Leodau) [reply doc/2/api/controllers/memory-storage/mexecute/index.md:7]: Ooops, forgot the caps.

[the remark this answers is not in the document]
Comment (scottinet) [reply doc/2/api/controllers/memory-storage/mexecute/index.md:7]: Makes sense.

**Comment by Aschen** [comment]
More a new-feature IMHO

**Pull request merged**